### PR TITLE
Adds the functions for Fock state preparation

### PR DESCRIPTION
**Pull request by nquesada**
Adds functions for coefficients, covariance matrices and vectors of means for describing a Fock state in terms of Gaussians.

### CODE REVIEW

File: strawberryfields/backends/bosonicbackend/backend.py [hunk 1]
```diff
@@ -287,8 +285,26 @@ def coef(arr):
...
     def prepare_fock(self, n, r=0.0001):
         """ Prepares the arrays of weights, means and covs of a Fock state"""
-
-        pass
+        if r ** 2 > 1 / n:
+            raise ValueError("The parameter r**2={} is larger than n={}".format(r ** 2, n))
+        # A simple function to calculate the parity
+        parity = lambda n: 1 if n % 2 == 0 else -1
+        # All the means are zero
+        means = np.zeros([n + 1, 2])
```
Comment (GDauphinais) [inline strawberryfields/backends/bosonicbackend/backend.py:293]: ### 
```suggestion
        means = np.zeros((n + 1, 2))
```

File: strawberryfields/backends/bosonicbackend/backend.py [hunk 2]
```diff
@@ -284,8 +285,26 @@ def coef(arr):
...
+        weights = np.array(
+            [
+                (1 - n * (r ** 2)) / (1 - (n - j) * (r ** 2)) * comb(n, j) * parity(j)
+                for j in range(n + 1)
+            ]
+        )
+        weights = weights / np.sum(weights)
+        return [[weights], [means], [covs]]
 
         ## To be completed here
```
Comment (GDauphinais) [inline strawberryfields/backends/bosonicbackend/backend.py:309]: This line could be removed

**Review by GDauphinais — CHANGES_REQUESTED** [review]

File: strawberryfields/backends/bosonicbackend/backend.py (hunk 1 above)
Comment (nquesada) [reply strawberryfields/backends/bosonicbackend/backend.py:293]: It's OK with lists as well.

**Review by GDauphinais — APPROVED** [review]

File: strawberryfields/backends/bosonicbackend/backend.py [hunk 3]
```diff
@@ -284,10 +285,26 @@ def coef(arr):
 
     def prepare_fock(self, n, r=0.0001):
```
Comment (GDauphinais) [inline strawberryfields/backends/bosonicbackend/backend.py:286]: What would be a good r value for a given n ? 
What's the error made as a function of n and r ?

**Pull request merged**